core/src/main/java/org/infinispan/factories/impl/DynamicModuleMetadataProvider.java

```diff
@@ -0,0 +1,28 @@
+package org.infinispan.factories.impl;
+
+import org.infinispan.commons.util.Experimental;
+import org.infinispan.configuration.global.GlobalConfiguration;
+
+/**
+ * Modules implementing {@link org.infinispan.lifecycle.ModuleLifecycle} might need additional control over the created
+ * components. They can hook into the module metadata building process during startup by implementing this interface
+ * which allows them to register additional component metadata dynamically.
+ * <p>
+ * This interface is currently experimental, it does not constitute a public API and is only used for internal testing
+ * purposes.
+ *
+ * @author anistor@redhat.com
+ * @since 10.0
+ */
+@Experimental
```

Comment:
No need to tag it `@Experimental` since it's in a non-public impl package

Reply:
oh, but it is experimental ! :)  fixed.